### DISCUSSION TITLE
Unable to upload huge Notion export

### DISCUSSION THREAD

**Discussion by etoosamoe** [outline/outline#7487] [Self Hosting] [answered]
Hi! Just deployed self-hosted instance (with pretty standard Docker Compose). Using local storage. Files uploading to pages works fine.

Upload limits set up to 30gb:
```
FILE_STORAGE_UPLOAD_MAX_SIZE=30000000000
FILE_STORAGE_WORKSPACE_IMPORT_MAX_SIZE=30000000000
```

I use Nginx as reverse proxy with standard config including `client_max_body_size 0;`.
And now I have .zip archive with Notion export around 23.5gb size.

**First of all**, I couldn't found any way to import and apply this archive via API (lack of upload speed on home internet).
**Second**, after long upload process, I see immideately 502 Bad Gateway from Nginx and no logs in container logs, except `{"label":"http","level":"info","message":"  <-- POST /api/files.create"}`.
Also, I see this "files" like uploaded to /tmp dir inside a container, not to uploads (yeah, two times in a row):
```
nodejs@ae6e99c878af:/opt/outline$ ls -lah /tmp
total 44G
drwxrwxrwt 1 root   root   4.0K Aug 29 20:09 .
drwxr-xr-x 1 root   root   4.0K Aug 29 15:46 ..
-rw-r--r-- 1 nodejs nodejs  22G Aug 29 16:45 242ae521334f7c22132676600
-rw-r--r-- 1 nodejs nodejs  22G Aug 29 20:09 242ae521334f7c22132676601
```

**Comment by tommoor** [comment, marked as answer]
The maximum file size that nodejs can access is 2.4Gb, so I'm afraid an import of this size is never going to work in one go. You'll need to unzip the archive and import in parts.

**Reply by etoosamoe** [reply]
@tommoor thanks for the rocket fast answer. 
I'm going to try it, but could you somewhen update the documentation about it?
And the last question, am I right, there is no way to upload and apply it via API?

**Reply by tommoor** [reply]
> And the last question, am I right, there is no way to upload and apply it via API?

In what respect? You can use the API to do anything that the app can do – for example you could write a script to iterate through your Notion export and create the collections and documents in your installation.

https://www.getoutline.com/developers#tag/documents/POST/documents.import
